src/material/tabs/tab.ts

```diff
@@ -99,6 +102,12 @@ export class MatTab implements OnInit, OnChanges, OnDestroy {
   /** Classes to be passed to the tab mat-tab-body container. */
   @Input() bodyClass: string | string[];
 
+  /**
+   * Custom ID for the tab, overriding the auto-generated one by Material.
+   * Note that when using this input, it's your responsibility to ensure that the ID is unique.
+   */
+  @Input() id: string | null = null;
```

Comment:
Since this is a new input, and no one is currently depending on it, wouldn't it be better if it was a signal input?